src/common/transformations/src/transformations/common_optimizations/softmax_fusion.cpp

```diff
@@ -8,63 +8,74 @@
 #include <ngraph/pattern/op/wrap_type.hpp>
 #include <ngraph/rt_info.hpp>
 #include <openvino/opsets/opset6.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
 #include <vector>
 
 #include "itt.hpp"
 #include "transformations/utils/utils.hpp"
 
 ov::pass::SoftmaxFusion::SoftmaxFusion() {
     MATCHER_SCOPE(SoftmaxFusion);
-    auto data_pattern = pass::pattern::any_input(pattern::has_static_rank());
+
+    auto data_pattern = pass::pattern::any_input(pass::pattern::has_static_rank());
     auto reduce_max_axes_pattern = ngraph::pattern::wrap_type<opset6::Constant>();
     auto reduce_max_pattern = ngraph::pattern::wrap_type<opset6::ReduceMax>({data_pattern, reduce_max_axes_pattern});
     auto sub_pattern = ngraph::pattern::wrap_type<opset6::Subtract>({data_pattern, reduce_max_pattern});
-    auto exp_pattern = ngraph::pattern::wrap_type<opset6::Exp>({sub_pattern});
+
+    auto exp_input = std::make_shared<pattern::op::Or>(OutputVector{sub_pattern, data_pattern});
+    auto exp_pattern = ngraph::pattern::wrap_type<opset6::Exp>({exp_input});
+
     auto reduce_sum_axes_pattern = ngraph::pattern::wrap_type<opset6::Constant>();
     auto reduce_sum_pattern = ngraph::pattern::wrap_type<opset6::ReduceSum>({exp_pattern, reduce_sum_axes_pattern});
     auto div_pattern = ngraph::pattern::wrap_type<opset6::Divide>({exp_pattern, reduce_sum_pattern});
 
-    ov::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+    ov::matcher_pass_callback callback = [=](pass::pattern::Matcher& m) {
         if (transformation_callback(m.get_match_root()))
             return false;
 
         const auto& pattern_map = m.get_pattern_value_map();
+        const auto& pshape = pattern_map.at(data_pattern).get_partial_shape();
+        auto rank = pshape.rank().get_length();
```

Comment:
Please verify that rank is static

Reply:
we already ensured that data shape is static in pattern matcher above
https://github.com/openvinotoolkit/openvino/blob/4b10df4f1810327a15df10f6e118e21c0570d0f5/src/common/transformations/src/transformations/common_optimizations/softmax_fusion.cpp#L20